### ISSUE TITLE
How to modify field resolver result from directive

### ISSUE THREAD

**Issue by akaSybe** [ChilliCream/hotchocolate#357]
Hi!

I'm trying to implement UpperCase directive using HotChocolate

Here is example how it could be implemented in ApolloServer: https://www.apollographql.com/docs/graphql-tools/schema-directives.html#Uppercasing-strings

I feel that it could be done in Directive Middleware but can't figured it out

**Comment by akaSybe** [comment]
That's how I try to solve this task:

```
public class UpperCaseDirectiveType : DirectiveType<UpperCaseDirective> 
{
    protected override void Configure(IDirectiveTypeDescriptor<UpperCaseDirective> descriptor)
    {
        descriptor
            .Name("uppercase")
            .Location(DirectiveLocation.Field | DirectiveLocation.FieldDefinition)
            .Middleware(next => context => Do(context, next));
    }

    private static async Task Do(IDirectiveContext context, DirectiveDelegate next)
    {
        await next(context);

        var result = context.Result as string;
        if (result != null)
        {
            result = result.ToUpper();
            context.Result = result;
        }
    }
}
```

but after calling `await next(context)` I get `context.Result = null`.

I think it's key point to problem

**Comment by michaelstaib** [comment]
It should work.... just give me a second, I will try it with the aspnet example.

**Comment by michaelstaib** [comment]
So, that is a bug.... but you can use a workaround.

You can also shortcircut the middleware and force a resolve like the following:

```csharp
  descriptor.Middleware(next => async context =>
            {
                context.Result = (await context.ResolveAsync<string>()).ToUpperInvariant();
            });
```

But, it should have worked like you did it. I will issue a fix tonight.

**Comment by akaSybe** [comment]
Thank you for quick response, I'll wait for fix.

**Comment by michaelstaib** [comment]
Fixed with #358
